### ISSUE TITLE
Ray cloudpickle, python, ray version check incompatible with `python setup.py develop`

### ISSUE THREAD

**Issue by pcmoritz** [ray-project/ray#1252]
### Describe the problem

When initializing ray that was installed with `python setup.py develop` we get the following error:

```
Exception: Version mismatch: The cluster was started with:
    Ray: 0.2.2
    Ray location: ray/__init__.pyc
    Python: 2.7.14
    Cloudpickle: 0.4.0
This process on node 192.168.1.205 was started with:
    Ray: 0.2.2
    Ray location: /Users/pcmoritz/ray/python/ray/__init__.pyc
    Python: 2.7.14
    Cloudpickle: 0.4.0
```

### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OS
- **Ray installed from (source or binary)**: source
- **Ray version**: latest master
- **Python version**: 2.7
- **Exact command to reproduce**:

**Comment by pcmoritz** [comment]
Note that this only happens if ray is used from within the ray/ folder (e.g. running an example), so it is not super severe but we should nonetheless fix it to make the user experience smooth.

**Comment by robertnishihara** [comment]
I'm having trouble reproducing this with Python 3.

**Comment by atumanov** [comment]
I ran into the same issue as of commit 044548bcfff4af4fdf1150ba6e65501b55feaa94 . 

OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS
Ray installed from (source or binary): source
Ray version: 044548bcfff4af4fdf1150ba6e65501b55feaa94
Python version: 2.7.12
Exact command to reproduce:
* inside ipython session: `import ray; ray.init()`

**Comment by robertnishihara** [comment]
Does the error always happen? Or only if you are inside of `ray/python`?

**Comment by robertnishihara** [comment]
@pcmoritz @atumanov can you check that this problem is fixed by #1307?
